configuration_service/configuration_service.go

```diff
@@ -78,61 +96,51 @@ func (service ConfigurationService) IsDaemonProcessingRequests(ctx context.Conte
 	if err = authutils.CompareWithLatestBlockNumber(big.NewInt(int64(auth.CurrentBlock))); err != nil {
 		return err
 	}
 
+	signerFromMessage, err := authutils.GetSignerAddressFromMessage(service.getMessageBytes(prefix, auth.CurrentBlock), auth.GetSignature())
+	if err != nil {
+		log.Error(err)
+		return err
+	}
 	//Check if the Signature is Valid and Signed accordingly
-	if err = authutils.VerifySigner(service.getMessageBytes(prefix, auth.CurrentBlock),
-		auth.GetSignature(), blockchain.HexToAddress(service.address)); err != nil {
+	if err = service.checkAuthenticationAddress(*signerFromMessage); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (service ConfigurationService) checkAuthenticationAddress(address string) error {
+func (service ConfigurationService) checkAuthenticationAddress(signer common.Address) error {
 
-	if !common.IsHexAddress(service.address) {
-		return fmt.Errorf("invalid hex address specified/missing for configuration 'authentication_address' ,this is a mandatory configuration required to be set up manually for remote updates")
-	}
-	if !common.IsHexAddress(address) {
-		return fmt.Errorf("%v is an invalid hex Address", address)
-	} else if strings.Compare(service.address, address) != 0 {
-		return fmt.Errorf("unauthorized access, %v is not authorized", address)
+	for _, user := range service.authenticationAddressList {
+		if user == signer {
+			return nil
+		}
 	}
+	return fmt.Errorf("unauthorized access, %v is not authorized", signer.Hex())
+
 	return nil
 }
 
 //You will be able to start the Daemon without an Authentication Address for now
-//but without Authentication address , you cannot use the operator UI functionality
+//but without Authentication authenticationAddressList , you cannot use the operator UI functionality
 func NewConfigurationService(messageBroadcaster *MessageBroadcaster) *ConfigurationService {
 	service := &ConfigurationService{
-		address: config.GetString(config.AuthenticationAddress),
-		broadcast:messageBroadcaster,
+		authenticationAddressList: getAuthenticationAddress(),
+		broadcast:                 messageBroadcaster,
 	}
-	authAddress := config.GetString(config.AuthenticationAddress)
-	//Make sure the address is a valid Hex Address
-	if !common.IsHexAddress(authAddress) {
-		service.address = ""
-		log.Errorf("invalid hex address specified/missing for 'authentication_address' in configuration , you cannot make remote update to current configurations")
-	}
-
 	return service
 }
 
-//Message format has been agreed to be as the below ( prefix,block number,and authenticating address)
-func (service *ConfigurationService) getMessageBytes(prefixMessage string, blocknumber uint64) []byte {
+//Message format has been agreed to be as the below ( prefix,block number,and authenticating authenticationAddressList)
+func (service *ConfigurationService) getMessageBytes(prefixMessage string, blockNumber uint64) []byte {
 	message := bytes.Join([][]byte{
```

Comment:
service.address is not used in the message?

Reply:
Not required , you derive the address from the signature ,